Reduce default mempool size in -blocksonly mode (#26471)
bitcoin/bitcoin: doc/release-notes-26471.md

Updated settings
----------------

> - Previously running with the `-blocksonly` option effectively disabled the

Reviewer: ```suggestion
- Previously running with the `-blocksonly` option all but disabled the
```

Author: Personally I feel "effectively disabled" is cleared here, so I've left it as-is for now (but happy to be persuaded)

Reviewer: The definition "actually but not officially or explicitly" captures "effectively" quite well for me, and in that sense I think is wrong here. But won't comment further, not that important.

Author: Removed in new version

  mempool but continued to allocate 300MB of cache for it, which would be

>   shared with the dbcache. This might lead to unexpected resource usage of
>   nodes running with this option during for example IBD.

Reviewer: ```suggestion
  shared with the dbcache. This might lead to nodes running with this
  option using more resources than anticipated, especially during IBD.
```

Author: Done

  In this release running with `-blocksonly` will only allocate 5MB to the
  mempool by default, permitting the use-case of relaying transactions from

>   whitelisted peers, but avoiding unexpected resource usage.

Reviewer: ```suggestion
  whitelisted peers, but avoiding unexpected resource usage. It reduces the
  available `dbcache` size by 295MB, impacting performance especially during
  IBD.
```

Author: Done

Updated settings
----------------

- Previously running with the `-blocksonly` option effectively disabled the
  mempool but continued to allocate 300MB of cache for it, which would be
  shared with the UTXO cache. This might lead to nodes running with this
  option using more resources than anticipated, especially during IBD.

  In this release running with `-blocksonly` will only allocate 5MB to the
  mempool by default, permitting the use-case of relaying transactions from
  whitelisted peers, but avoiding unexpected resource usage. It reduces the

>   available `dbcache` size by 295MB, impacting performance especially during

Reviewer: Perhaps need to add here that this is only true if `-maxmempool` is not set? We say it in other places, but better to not assume everyone reads everything, so seems like a safe option.

>   IBD.

Author: I agree that we don't  need to mention the whitelist relay behaviour, but its apparently something some people (might) [have been doing](https://github.com/bitcoin/bitcoin/pull/9569#issuecomment-273338305).

Anyway, removed/
